### ISSUE TITLE
Kubelet permissions check on Windows does not work if running as a LocalSystem

### ISSUE THREAD

**Issue by marosset** [kubernetes/kubernetes#108140]
### What happened?

If kubelet is configured to run as a service using the `LocalSystem` service account check to verify the kubelet is running with elevated privileges fails with the follow error message

```cmd
E0127 18:48:07.328394    4752 server.go:287] "kubelet running with insufficient permissions" err="error while checking admin group membership: error retrieving group ids: The user name could not be found."
```

### What did you expect to happen?

Running the kubelet as a Windows serrvier using `LocalSystem` should not report kubelet is running with insufficient permissions.

### How can we reproduce it (as minimally and precisely as possible)?

On a windows node run kubelet.exe as a service using the `LocalSystem` account.

### Anything else we need to know?

https://github.com/kubernetes/kubernetes/blob/4033355d292505eaed05655bd91f4ea9edc7ef27/cmd/kubelet/app/server_windows.go#L29-L49

This logic checks to see if the currently running user is in the `BUILTIN\Administrators` group.
This is the correct logic if kubelet is running as a user account but does not work correctly when running as one of the built-in service accounts (Local System, Network Service, or Local Service).

Here the call to get the group Ids for a given user is failing because saying that the specified username does not exist.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by marosset** [comment]
/sig windows node
/area kubelet

Related to https://github.com/kubernetes/kubernetes/pull/96616

**Comment by marosset** [comment]
I think we can probably just https://pkg.go.dev/golang.org/x/sys/windows#GetCurrentProcessToken and check to see if token.IsElevated is true.
This does not account for the case where kubelet is not running elevated but is running with an account that _could_ elevate but I don't think we care about that (because we'd need to relaunch kubelet as elevated).
I think we just need to check/ensure the current process _is_ running running as elevated...

**Comment by marosset** [comment]
/triage accepted
The current behavior does not work for some common scenarios.

I've got a PR open for a fix (#108146) that does need node reviews. 
Thanks!
